lib/uri/common.rb

```diff
@@ -91,7 +91,7 @@ def self.for(scheme, *arguments, default: Generic)
     const_name = scheme.to_s.upcase
 
     uri_class = INITIAL_SCHEMES[const_name]
-    if !uri_class && !const_name.empty? && Schemes.const_defined?(const_name, false)
+    if !uri_class && !const_name.empty? && /\A[A-Z][A-Z0-9_]+\z/.match?(const_name) && Schemes.const_defined?(const_name, false)
```

Comment:
`&& !const_name.empty?` can be removed with this change.

> `/\A[A-Z][A-Z0-9_]+\z/.match?(const_name)`

Should it be `/\A[A-Z]\w*\z/.match?(const_name)` rather (allow lowercase letters from the 2nd character, single upper character allowed)?

Reply:
Definitely, I shouldn't have implemented this this late. Will fix.